File: Chapter06/AerialScrew.lua
-- function AerialScrew
function AerialScrew()
	-- configure the light sensor on input port 3
	nxt.InputSetDigi0(3)
	nxt.InputSetDirOutDigi0(3)
	-- configure motor A to run in "brake" mode
	nxt.OutputSetMode(1,2)
	-- define a variable that holds the present rotation speed
	-- initial value is 20
	local rotationSpeed = 20
	-- loop until rotation speed is maximum (100)
	while rotationSpeed < 100 do
		-- set the power of the helix's motor to the rotation speed
		-- 32 means "run mode"
		nxt.outputSetSpeed(1, 32, rotationSpeed, 0)
		-- loop until light sensor detects the passing spill grip
		while nxt.InputGetRawAd(3) >= 50
			-- do nothing
		end
		-- increment the rotation speed
		rotationSpeed = rotationSpeed + 1
	end
	-- run the motor with maximum power
	nxt.outputSetSpeed(1, 32, 100, 0)
end

-- Now run the AerialScrew function
AerialScrew()

File: Chapter06/AerialScrew.java
package org.nxtdavinci.aerialscrew;

import lejos.nxt.*;

public class AerialScrew {
	public static void main(String[] args) {
		// configure the light sensor on input port 3
		LightSensor lightSensor = new LightSensor(Port.S3);
		// define a variable that holds the present rotation speed
		// initial value is 20
		int rotationSpeed = 20;
		// loop until rotation speed is maximum (100)
		while(rotationSpeed < 100) {
			// set the power of the helix's motor to the rotation speed
			Motor.A.setSpeed(rotationSpeed);
			// loop until light sensor detects the passing spill grip
			do {
			// run the motor forward
			Motor.A.forward();
			// check the light sensor
			} while(lightSensor.readValue() >= 50);
			// increment the rotation speed
			rotationSpeed++;
		}
		// run the motor with maximum power
		Motor.A.setSpeed(100);
		while(true) {
			Motor.A.forward();
		}
	}
}